### DISCUSSION TITLE
domain folder structure

### DISCUSSION THREAD

**Discussion by mkalinski93** [nuxt/nuxt#16364] [Questions]
Hello,

I just wanted to ask if it´s possible to change the folder structure in a domain-like manner.
The pages and components folder can become quiet large, so it could be useful to structure the application a bit different.

Currently, I´m trying to achieve the following structure:

modules
 - domainA
 - - pages
 -  - components
 -  - composables
 - domainB
 -  - pages
 -  - components
 -  - composables

components and composables seems to be working.
I create for each domain a modules.ts file that defines the domain module

e.g: 

```typescript
export default defineNuxtModule({
    hooks:{
        "components:dirs"(dirs) {
            // Add ./components dir to the list
            dirs.push({
                path: path.resolve(__dirname, "./components"),
            });
        },
        "imports:dirs"(dirs) {
         dirs.push(path.resolve(__dirname, "./composables"));
        }
    }
})
```

for pages, I need to manually specify the route like this

```typescript
"pages:extend"(pages){
            pages.push({
                name:'user',
                path: '/users/',
                file: path.resolve(__dirname, './pages/index.vue')
            })
        }
```

Can I auto extend the pages routes in the following manner: 

modules/users/index.vue => /users
modules/users/[id].vue => /users/1
modules/users-[group]/[id].vue => /users-agent/1

I have found a similar [question nuxt/framework#5683](https://github.com/nuxt/framework/discussions/5683) but it has been remained unanswered

**Comment by TheAlexLichter** [comment]
Yes. But what you want is the `extends` feature! See https://github.com/nuxt/framework/issues/3222 😋

**Reply by IsraelOrtuno** [reply]
@mkalinski93 you should watch @manniL's talk at NuxtNation.com which I guess will be available online somewhere in the upcoming days. A solid example of what can be done with the `extends` feature and what to expect from it.

**Reply by TheAlexLichter** [reply]
Thanks @IsraelOrtuno ☺️ 

[Link to slides and examples](https://www.lichter.link/nuxt-nation-2022-repo/)

**Reply by mkalinski93** [reply]
That's great news! I'm really excited to see it 👍
